src/server/services/pipeline-in-progress/processData.js

```diff
@@ -0,0 +1,169 @@
+ * @param {dict} conditionSet used to validate readings (minVal, maxVal, minDate, maxDate, interval, maxError)
+ */
+
+function processData(rows, meterID, isCumulative, cumulativeReset, resetStart = '0:00:00.000', 
+					resetEnd = '23:59:99.999', readingRepetition, onlyEndTime = false, Tgap, Tlen, conditionSet) {
+
+	// If processData is succesfully finished then return result = [R0, R1, R2...RN]
+	const result = [];
+
+	let errMsg = '';
+
+	let meterReading = 0;
+	let meterReading1 = 0;
+	let meterReading2 = 0;
+
+	let startTimestamp = moment(0);
+	let endTimestamp = moment(0);
+
+	let currentReading = new Reading(meterID, -1, startTimestamp, endTimestamp);
+	let prevReading = currentReading;
+	let readingOK = true;
+	for (let index = readingRepetition; index <= rows.length; ++index) {
+		// To read data where same reading is repeated. Like E-mon D-mon meters
+		if ((index - readingRepetition) % readingRepetition === 0) {
+			if (onlyEndTime) {
+				// The startTimestamp of this reading is the endTimestamp of the previous reading
+				startTimestamp = prevReading.endTimestamp;
+				endTimestamp = moment(rows[index - readingRepetition][1], ['YYYY/MM/DD HH:mm', 'MM/DD/YYYY HH:mm']);
+			}
+			else {
+				startTimestamp = moment(rows[index - readingRepetition][1], ['YYYY/MM/DD HH:mm', 'MM/DD/YYYY HH:mm']);
+				endTimestamp = moment(rows[index - readingRepetition][2], ['YYYY/MM/DD HH:mm', 'MM/DD/YYYY HH:mm']);
+			}
+			if (isCumulative && isFirst(prevReading.endTimestamp)) {
+				readingOK = false;
+				errMsg = 'The first reading must be dropped when dealing with cumulative data.';
+			}
+			if (onlyEndTime && isFirst(prevReading.endTimestamp)) {
+				readingOK = false;
+				errMsg = 'The first reading must be dropped when dealing only with endTimestamps.';
+			}
+			if (readingOK && startTimestamp.isSameOrAfter(endTimestamp)) {
+				readingOK = false;
+				errMsg = 'The reading end time is not after the start time.';
+				if (onlyEndTime) {
+					errMsg += ' The start time came from the previous readings end time.';
+				}
+			}
+			if (readingOK) {
+				// Check that startTimestamp is not before the previous endTimestamp
+				if (onlyEndTime && endTimestamp.isSameOrBefore(prevReading.endTimestamp)) {
+					readingOK = false;
+					errMsg = 'The reading is not after the previous reading with only end time given so we must drop the reading.';
+				}
+				else if (startTimestamp.isBefore(prevReading.endTimestamp)) {
+					if (isCumulative) {
+						readingOK = false;
+						errMsg = 'The reading start time is before the previous endTime and cumulative so OED cannot use this reading.'
+					}
+					else if (!isFirst(prevReading.endTimestamp)) {
+						//Only treat this as a warning since the readings may be sent in a different order.
+						errMsg = 'The current reading startTime is not after the previous reading\'s end time. This is expected on the first reading ever.';
```

Comment:
You're right. I've changed it so that the second sentence was removed and the user will instead see:

"The current reading startTime is not after the previous reading's end time. Note this is treated only as a warning since readings may be sent out of order"